### PR TITLE
ORIGIN-670: Add Support for Interconnectors

### PR DESCRIPTION
**Pull request by jebbin-joseph**
[ORIGIN-670](https://auroraenergy.atlassian.net/browse/ORIGIN-670)

[ORIGIN-670]: https://auroraenergy.atlassian.net/browse/ORIGIN-670?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docsite/docs/origin_sdk/OriginSession.md [hunk 1]
```diff
@@ -313,7 +313,7 @@ object).
 ```python
 @access_next_data_key_decorator
 def get_commodities(scenario_id: str,
-                    native_units_flag: bool = None,
+                    native_units_flag: Optional[bool] = None,
```
Comment (jebbin-joseph) [inline docsite/docs/origin_sdk/OriginSession.md:316]: This change was pending from the last update to the file. The argument is Optional in the function.

File: docsite/docs/origin_sdk/OriginSession.md [hunk 2]
```diff
@@ -341,7 +341,7 @@ def update_commodity_price(scenario_id: str,
                            commodity: str,
                            regions: List[str],
                            transform: List[Transform],
-                           native_units_flag: bool = None)
+                           native_units_flag: Optional[bool] = None)
```
Comment (jebbin-joseph) [inline docsite/docs/origin_sdk/OriginSession.md:344]: Same here.

File: src/origin_sdk/OriginSession.py [hunk 3]
```diff
@@ -626,3 +627,104 @@ def change_base_commodities_assumptions(
...
+            self.inputs_service_graphql_url,
+            input_query.get_interconnectors_regions_gql(),
+            variables,
+        )
+
+        all_connections = list(
+            map(lambda x: x["ends"], interconnector_ends["getInterconnectors"])
+        )
+
+        connection_dict = defaultdict(set)
```
Comment (frisbee09) [inline src/origin_sdk/OriginSession.py:654]: What is this

Comment (frisbee09) [reply src/origin_sdk/OriginSession.py:654]: Ok, I looked up docs. Comment this please it's uncommon python but v. cool

**Comment by frisbee09** [comment]
You need to also rebuild and redeploy the documentation

**Review by frisbee09 — APPROVED** [review]

**Pull request merged**